src/bosh-softlayer-cpi/action/create_vm.go

```diff
@@ -77,10 +77,18 @@ func (a CreateVMAction) Run(agentID string, stemcellCID StemcellCID, cloudProps
 func (a CreateVMAction) updateCloudProperties(cloudProps *VMCloudProperties) {
 	a.vmCloudProperties = cloudProps
 
-	if len(cloudProps.BoshIp) == 0 || cloudProps.Baremetal {
-		a.vmCloudProperties.VmNamePrefix = cloudProps.VmNamePrefix
+	if cloudProps.NotDeployedByDirector {
```

Comment:
I dislike the negative name here. So for me, something like: `if !cloudProps.DeployedByDirector { ...` reads better. But perhaps that's just me.